handlers/common.go

```diff
@@ -31,10 +32,58 @@ func PingNoOp(event *events.Event, apiClient *client.RancherClient) error {
 	return nil
 }
 
+func acquire_networked_global_persistent_lock(resourceId string) error {
+	return nil
+}
+
+func release_lock(lock string) {
+	return
+}
+
 func getMachineDir(machine *client.Machine) (string, error) {
 	return getBaseMachineDir(machine.ExternalId)
 }
 
+func sync_resource(machine *client.Machine) error {
```

Comment:
It would be nice to cleanup after ourselves, but I want to keep that logic as simple as possible. Specifically, we shouldn't introduce any logic that requires cattle or gms to "know" about specific instances of gms running.

Also, I think this cleanup could be relatively low priority. Meaning, we could ship this without the cleanup code.

All that said, how about this for a solution: we have a cleanup function that blows away the top-level machine directory so that all state (for all machines) is deleted. In the short term, we could call that on startup of gms (before gms starts processing any events from cattle). In the longer term, cattle could send a periodic event to gms telling it to run the cleanup function (or gms could just do it on a self-timer, without cattle prompting it). The important thing would be ensuring that we lock gms out of processing any events until the cleanup is completed (so we don't blow away any machines that are mid-update).